### ISSUE TITLE
Can't setup the integration

### ISSUE THREAD

**Issue by Mattat01** [vinteo/hass-opensprinkler#128]
Hi,
I'm getting an unexpected exeption when entering the url, password, mac and name for the openSprinkler:

> Logger: custom_components.hass_opensprinkler.config_flow
> Source: custom_components/hass_opensprinkler/config_flow.py:51
> Integration: OpenSprinkler
> First occurred: 5:58:38 PM (5 occurrences)
> Last logged: 6:09:48 PM
> 
> Unexpected exception
> Traceback (most recent call last):
>   File "/config/custom_components/hass_opensprinkler/config_flow.py", line 51, in async_step_user
>     await self.hass.async_add_executor_job(controller.refresh)
>   File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
>     result = self.fn(*self.args, **self.kwargs)
>   File "/usr/local/lib/python3.7/site-packages/pyopensprinkler/__init__.py", line 196, in refresh
>     self._refresh_state()
>   File "/usr/local/lib/python3.7/site-packages/pyopensprinkler/__init__.py", line 214, in _refresh_state
>     (_, content) = self.request("/ja")
>   File "/usr/local/lib/python3.7/site-packages/pyopensprinkler/__init__.py", line 144, in request
>     (resp, content) = self.request_http(url)
>   File "/usr/local/lib/python3.7/site-packages/pyopensprinkler/__init__.py", line 55, in newFunction
>     return f(*args, **kw)
>   File "/usr/local/lib/python3.7/site-packages/backoff/_sync.py", line 94, in retry
>     ret = target(*args, **kwargs)
>   File "/usr/local/lib/python3.7/site-packages/pyopensprinkler/__init__.py", line 170, in request_http
>     (resp, content) = self._http_client.request(url, "GET", headers=headers)
>   File "/usr/local/lib/python3.7/site-packages/httplib2/__init__.py", line 1994, in request
>     cachekey,
>   File "/usr/local/lib/python3.7/site-packages/httplib2/__init__.py", line 1651, in _request
>     conn, request_uri, method, body, headers
>   File "/usr/local/lib/python3.7/site-packages/httplib2/__init__.py", line 1557, in _conn_request
>     conn.connect()
>   File "/usr/local/lib/python3.7/site-packages/httplib2/__init__.py", line 1236, in connect
>     raise socket_err
>   File "/usr/local/lib/python3.7/site-packages/httplib2/__init__.py", line 1210, in connect
>     self.sock.connect((self.host, self.port) + sa[2:])
> OSError: [Errno 113] Host is unreachable

I am certain the url and pw and mac address are correct as I can log into the openSprinker interface on my browser and phone with those details. I've tried lower and upper case for letters in the mac address, as sometimes that matters, and separated the mac address with : - and no separators.

I have App Version 2.2.0, Firmware 2.1.8 (2) and Hardware Version OSPi.

The form just resets without doing anything. I presume that the data should be stored in strings.json but adding it there manually didn't work (probably unsurprisingly).

Do you know what my be going wrong?

Thanks,
Matthew

**Comment by vinteo** [comment]
I am sorry but that looks like a network issue. Can HA and OpenSprinkler see each other on your network? You can try install the beta version to see if that makes a difference

**Comment by Mattat01** [comment]
I believe HASS should be able to contact OpenSprinkler. They are on the same wifi and I can connect to OpenSprinkler with other devices on the same wifi. It also used to work on an earlier version (when HASS was something like 0.90), but I noticed it was no longer turning on stations so I tried upgrading to the current version. Do you know a way to explicitly check if they can talk?

My first error today was that I didn't notice that you changed the folder name. Saddly I fixed that and it still isn't working.

Even if I type rubbish into the url and pw fields I get the same error. I was trying to trigger the cannot_connect or invalid_auth error.

Where is the beta code stored? I'm happy to try that.

**Comment by vinteo** [comment]
https://github.com/vinteo/hass-opensprinkler/pull/123

**Comment by Mattat01** [comment]
Thanks. I uninstalled it and reinstalled using HACS. I found the beta through that. with the release and beta version I'm now getting the can't connect error - "Giving up _request_http(...) after 3 tries (pyopensprinkler.OpenSprinklerConnectionError: Cannot connect to controller)".
I can't imagine why it can't connect as I can connect from every other device, but I'll have to try more in the morning.
Thanks for the help.

**Comment by Mattat01** [comment]
Nope, I can't work it out. Still getting the can't connect error with the beta version. Upgraded to Hassio 0.113.3 but it didn't change anything. Do you know if there is a manual way to configure it? Where does the url/password get stored?

**Comment by vinteo** [comment]
Just checking you are using the full url for the config right ie `http://[ip]:[port]`. Note that the default port of OS is `8080`

**Comment by Mattat01** [comment]
Yes I am. In my case it is http://10.0.0.102:8080, the same as I use in a browser to access it, and the same password too. I have the hardware that needs the mac address too, but whether I add it or not does not matter at this stage (I've tried to see if I can get the 'MAC address required' error, but it obviously doesn't connect to check if it needs it).

**Comment by vinteo** [comment]
The password should not be md5'ed in the integration setup config (if you are putting it in the browser it would be). But it should have given a wrong password error.

**Comment by Mattat01** [comment]
Oh. It just worked... No idea what the problem was. I noticed that my powerwall inegration had also dropped of so I deleted that integration, increased my logging records from 'critical' to 'warning' and started again. Several failures on that one and a few reboots and suddenly both work...

Sorry I can't determine what the cause was, or what fixed it. It obviously wasn't your end or the network but somethink blocking up hassio's guts so it had a spaz...

Thanks for working though it wiht me regardless, I've always been a fan of your integration and support!

Matthew
